pykep/python_udpla.cpp

```diff
@@ -39,20 +41,44 @@ python_udpla::python_udpla(py::object obj) : m_obj(std::move(obj))
 };
 
 // Mandatory methods
-[[nodiscard]] std::array<std::array<double, 3>, 2> python_udpla::eph(const kep3::epoch &ep) const
+[[nodiscard]] std::array<std::array<double, 3>, 2> python_udpla::eph(double mjd2000) const
 {
-    auto bf = pykep::callable_attribute(m_obj, "eph");
-    if (bf.is_none()) {
+    auto isthere = pykep::callable_attribute(m_obj, "eph");
+    if (isthere.is_none()) {
         pykep::py_throw(PyExc_NotImplementedError, ("the eph() method has been invoked, but it is not implemented "
                                                     "in the user-defined Python planet '"
                                                     + pykep::str(m_obj) + "' of type '" + pykep::str(pykep::type(m_obj))
                                                     + "': the method is either not present or not callable")
                                                        .c_str());
     }
-    return py::cast<std::array<std::array<double, 3>, 2>>(m_obj.attr("eph")(ep));
+    return py::cast<std::array<std::array<double, 3>, 2>>(m_obj.attr("eph")(mjd2000));
 }
 
 // Optional methods
+[[nodiscard]] std::vector<double> python_udpla::eph_v(const std::vector<double> &mjd2000s) const
+{
+    auto has_eph_v = !pykep::callable_attribute(m_obj, "eph_v").is_none();
+    if (has_eph_v) {
+        auto ret = py::cast<py::array_t<double>>(m_obj.attr("eph_v")(mjd2000s));
+        auto retval = py::cast<std::vector<double>>(ret.attr("flatten")());
+        return retval;
+    } else {
+        // We simply call a for loop.
```

Comment:
I think it would be worth it to avoid the repetition of this code by putting the default ``eph_v()`` implementation with the loop into a little detail helper in ``planet.cpp``, and invoke it here and in ``planet.hpp`` (this will also reduce compilation times).

Reply:
And I pass `this` to the helper?
```
return kep3::detail::my_little_helper(this, mjd2000s)
```

Reply:
no can do. I think we need to pass a function pointer to an eph method?